nixos/modules/programs/z.nix

```diff
@@ -0,0 +1,109 @@
+with builtins;
+{ config, lib, pkgs, ... }:
+  let
+    l = lib; p = pkgs; t = l.types;
+
+    make-bool-options =
+      mapAttrs
+        (_: description:
+           l.mkOption
+             { type = t.bool;
+               inherit description;
+               default = true;
+             }
+        );
+
+    make-str-options =
+      mapAttrs
+        (_: description:
+           l.mkOption
+             { type = t.nullOr t.str;
+               inherit description;
+               default = null;
+             }
+        );
+
+    bool-options =
+      { resolve-symlinks = "resolve symlinks";
+
+        prompt-command =
+          ''
+          Automatically have z handle the PROMPT_COMMAND/precmd.
+
+          Note: if you override your PROMPT_COMMAND after z does, it will override the changes z makes, and z will not work. You can do something like
+
+          export PROMPT_COMMAND="my-prompt-command $PROMPT_COMMAND"
+
+          to fix this.
+          '';
```

Comment:
```suggestion
        prompt-command = ''
          Automatically have z handle the PROMPT_COMMAND/precmd.
          Note: if you override your PROMPT_COMMAND after z does, it will override the changes z makes, and z will not work. You can do something like
          export PROMPT_COMMAND="my-prompt-command $PROMPT_COMMAND"
          to fix this.
        '';
```